addons/website/models/ir_ui_view.py

```diff
@@ -94,41 +94,15 @@ def write(self, vals):
 
         return True
 
-    @api.multi
-    def _get_specific_views(self):
-        """ Given a view, return a record set containing all the specific views
-            for that view's key.
-            If the given view is already specific, it will also return itself.
-        """
-        self.ensure_one()
-        domain = [('key', '=', self.key), ('website_id', '!=', False)]
-        return self.with_context(active_test=False).search(domain)
-
-    def _load_records_write(self, values):
-        """ During module update, when updating a generic view, we should also
-            update its specific views (COW'd).
-            Note that we will only update unmodified fields. That will mimic the
-            noupdate behavior on views having an ir.model.data.
-        """
-        if self.type == 'qweb' and not self.website_id:
-            # Update also specific views
-            for cow_view in self._get_specific_views():
-                authorized_vals = {}
-                for key in values:
-                    if cow_view[key] == self[key]:
-                        authorized_vals[key] = values[key]
-                cow_view.write(authorized_vals)
-        super(View, self)._load_records_write(values)
-
-    def _load_records_create(self, values):
-        """ During module install, when creating a generic child view, we should
+    def _validate_custom_views(self, model):
+        """ When creating a generic child view, we should
             also create that view under specific view trees (COW'd).
             Top level view (no inherit_id) do not need that behavior as they
             will be shared between websites since there is no specific yet.
         """
-        records = super(View, self)._load_records_create(values)
-        for record in records:
-            if record.type == 'qweb' and record.inherit_id and not record.website_id and not record.inherit_id.website_id:
+        if self.pool.load_records_create_duplicated_views:
```

Comment:
@fja-odoo 

Can't we use load_records_create_duplicated_views to stores ids to replay, and avoid to check for each view one by one ? because on production database, it will be huge.

And get specific_parent_views with only one request, and aggregate the specific view.

Are we sure that the write on a view in the middle of the tree, will not duplicate again all the childs ?

Reply:
We can store the ids of the views being created to avoid multiple useless search yes. Just know that specific views created by this and deleted by the user will not be recreated on update. I do not think this is an issue as it is the current behavior.

I will try to aggragate the specific view.

Duplication of childs If we create only from load_records_create_duplicated_views will not be possible. In the current state I do not know, thats another good point to use load_records_create_duplicated_views to store the ids.